packages/test/local-server-stress-tests/src/test/localServerStressOrderSequentially.spec.ts

```diff
@@ -0,0 +1,97 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { done, isOperationType, takeAsync } from "@fluid-private/stochastic-test-utils";
+
+import { makeGenerator, reducer, saveFailures, type StressOperations } from "../baseModel.js";
+import {
+	convertToRealHandles,
+	covertLocalServerStateToDdsState,
+	DDSModelOpGenerator,
+	loadAllHandles,
+	validateConsistencyOfAllDDS,
+	type DDSModelOp,
+	type OrderSequentially,
+} from "../ddsOperations.js";
+import {
+	createLocalServerStressSuite,
+	LocalServerStressModel,
+	type LocalServerStressState,
+} from "../localServerStressHarness.js";
+
+const orderSequentiallyReducer = async (
+	state: LocalServerStressState,
+	op: OrderSequentially,
+) => {
+	const { baseModel, taggedHandles } = await loadAllHandles(state);
+	const ddsState = await covertLocalServerStateToDdsState(state);
+	const rollbackError = new Error("rollback");
+	try {
+		state.datastore.orderSequentially(() => {
+			for (const o of op.operations) {
+				baseModel.reducer(ddsState, convertToRealHandles(o, taggedHandles));
+				if (op.rollback) {
+					throw rollbackError;
+				}
+			}
+		});
+	} catch (error) {
+		if (error !== rollbackError) {
+			throw error;
+		}
+	}
+};
+
+describe("Local Server Stress with rollback", () => {
+	const model: LocalServerStressModel<StressOperations | OrderSequentially> = {
+		workloadName: "rollback",
+		generatorFactory: () =>
+			takeAsync(
+				100,
+				makeGenerator<OrderSequentially>([
+					[
+						async (state) => {
+							const operations: DDSModelOp[] = [];
+							/**
+							 * unfortunately we can't generate more than a single op here, as each op is generated off
+							 * the current state, and if we generate multiple ops it can result in earlier ops invaliding
+							 * the constrains necessary for later ops. for example, an earlier op might delete a sub-directory
+							 * which a later op sets a key in, but the state and generator don't know that will happen.
+							 */
+							const op = await DDSModelOpGenerator(state);
+							if (op !== done) {
+								operations.push(op);
```

Comment:
When would the operation be returned to us already `done`?

Reply:
when the generator is empty. it won't really happen, but the typing thinks its possible